test/new_relixir/plug/exception_test.exs

```diff
@@ -0,0 +1,54 @@
+defmodule ExceptionTest do
+  use ExUnit.Case
+  use Plug.Test
+
+  defmodule TestException do
+    defexception plug_status: 403, message: "oops"
+  end
+
+  defmodule ErrorRaisingPlug do
+    defmacro __using__(_env) do
+      quote do
+        def call(conn, _opts) do
+          raise Plug.Conn.WrapperError, conn: conn,
+          kind: :error, stack: System.stacktrace,
+          reason: TestException.exception([])
+        end
+      end
+    end
+  end
+
+  defmodule TestPlug do
+    use NewRelixir.Plug.Exception
+    use ErrorRaisingPlug
+  end
+
+  defmodule FakeReporter do
+    def record_error(transaction, exception) do
+      send self(), {:record_error, {transaction, exception}}
```

Comment:
I'm not sure why you're doing this? Is there a contract to respond to a message or am I missing something or ???

Reply:
I think it's just so the test can verify the the reporters record error function was called with the correct arguments. Logging something and capturing the log would be another way to accomplish this

Reply:
Exactly, this is modelled after plugsnag

Reply:
Thanks. LGTM